tests/unit/test_version_check.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+import unittest
+from unittest.mock import patch
+
+from ops.testing import Harness
+
+from charm import MongodbOperatorCharm
+from version_check import DEPLOYMENT_TYPE, VERSION_CONST, NoVersionError
+
+from .helpers import patch_network_get
+
+CHARMHUB_DEPLOYMENT = "ch"
+LOCAL_DEPLOYMENT = "local"
+RELATION_TO_CHECK_VERSION = "sharding"
+CHARM_VERSION = 123
+VALID_VERSION = CHARM_VERSION
+INVALID_VERSION = 456
+APP_0 = "APP_0"
+APP_1 = "APP_1"
+APP_2 = "APP_2"
+
+
+class TestCharm(unittest.TestCase):
+    @patch("charm.get_charm_revision")
+    @patch_network_get(private_address="1.1.1.1")
```

Comment:
Do we need to patch network_get?

Reply:
I'm not sure. In general most unit tests need it. But probably not needed here. I can remove it next time I revisit this file like we discussed